packages/flutter/lib/src/widgets/conduit.dart

```diff
@@ -0,0 +1,375 @@
+
+  @override
+  RenderConduitAnchorWidget get widget => super.widget as RenderConduitAnchorWidget;
+
+  @override
+  _RenderFollowerProxyBox get renderObject => super.renderObject as _RenderFollowerProxyBox;
+
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+
+    final _PendingRenderObjectManifest pending = widget.key._manifest.content;
+
+    if (pending == null)
+      return;
+
+    renderObject.leader = pending.leaderRenderBox;
+    final RenderBox pendingRenderBox = pending.renderObject as RenderBox;
+    insertChildRenderObject(pendingRenderBox, null);
+    widget.key._manifest.content = null;
+  }
+
+  @override
+  void unmount() {
+    renderObject.leader = null;
+    super.unmount();
+  }
+
+  @override
+  void insertChildRenderObject(RenderBox child, dynamic slot) {
+    assert(slot == null);
+    final RenderProxyBox oldRenderObject = renderObject;
+    assert(oldRenderObject.debugValidateChild(child));
+    renderObject.child = child;
+    assert(renderObject == oldRenderObject);
+  }
+
+  @override
+  void removeChildRenderObject(RenderObject child) {
+    final RenderProxyBox oldRenderObject = renderObject;
+    // This is more forgiving than a typical removeChildRenderObject implementation,
+    // since _RenderConduitAnchorElement detaches its remoteChild.renderObject in
+    // `Element.deactivate()`.
+    if (oldRenderObject.child != null) {
+      assert(oldRenderObject.child == child);
+      renderObject.child = null;
+    }
+    renderObject.leader = null;
+    assert(renderObject == oldRenderObject);
+  }
+}
+
+// A RenderProxyBox subclass that keeps its follower's depth greater than or equal
+// to its own depth, if its follower is not null.
+class _RenderLeaderProxyBox extends RenderProxyBox {
+  _RenderLeaderProxyBox([RenderBox child]) : super(child);
+
+  _RenderFollowerProxyBox __follower;
+  _RenderFollowerProxyBox get _follower => __follower;
+  set _follower(_RenderFollowerProxyBox value) {
+    if (value == _follower)
+      return;
+
+    assert(
+      (value == null) != (_follower == null),
+      '$this already has a follower: $_follower, new follower $value cannot be assigned.',
+    );
+    _follower?._leader = null;
+    __follower = value;
+    _follower?.leader = this;
+  }
+
+  @override
+  void redepthChildren() {
+    super.redepthChildren();
+    if (_follower != null)
+      redepthChild(_follower);
+  }
+}
+
+class _RenderFollowerProxyBox extends RenderProxyBox {
+  _RenderFollowerProxyBox([RenderBox child]) : super(child);
+
+  _RenderLeaderProxyBox _leader;
+  _RenderLeaderProxyBox get leader => _leader;
+  set leader(_RenderLeaderProxyBox value) {
+    if (value == leader)
+      return;
+
+    assert(
+      (value == null) != (leader == null),
+      '$this already has a leader: $leader, new leader $value cannot be assigned.',
+    );
+    leader?.__follower = null;
+    _leader = value;
+    leader?._follower = this;
```

Comment:
It's weird that there's one setter on each side and it sets the other side. It means it's not clear who owns these fields, the leader or the follower. I think it would be better if one of them (the leader) was in charge of the other. That way there would be no ambiguity about which one should be set.